### ISSUE TITLE
Loading issue when `.data` section is created

### ISSUE THREAD

**Issue by lxylxy123456** [UCDClassNitta/riscv-console#7]
If you replace `riscv-example/src/main.c` with the following program, you will see:
![图片](https://user-images.githubusercontent.com/9337403/105938731-a93aa900-600c-11eb-881b-965bfbb5a57f.png)

Expected:
![图片](https://user-images.githubusercontent.com/9337403/105938757-b5266b00-600c-11eb-94a1-4b1d9ebac24d.png)

It seems that if line 63 is commented out, the error disappears.

Here are output of `readelf -SW bin/riscv-console-example.strip`

63 commented out:
```
  [Nr] Name              Type            Addr     Off    Size   ES Flg Lk Inf Al
  [ 0]                   NULL            00000000 000000 000000 00      0   0  0
  [ 1] .text             PROGBITS        00000000 001000 000680 00  AX  0   0  4
  [ 2] .data             PROGBITS        70000000 002000 000023 00  WA  0   0  4
  [ 3] .sdata            PROGBITS        70000024 002024 000028 00  WA  0   0  4
  [ 4] .bss              NOBITS          7000004c 00204c 000004 00  WA  0   0  4
  [ 5] .comment          PROGBITS        00000000 00204c 000012 01  MS  0   0  1
  [ 6] .riscv.attributes RISCV_ATTRIBUTES 00000000 00205e 000025 00      0   0  1
  [ 7] .shstrtab         STRTAB          00000000 002083 00003e 00      0   0  1
```

63 not commented out:
```
  [Nr] Name              Type            Addr     Off    Size   ES Flg Lk Inf Al
  [ 0]                   NULL            00000000 000000 000000 00      0   0  0
  [ 1] .text             PROGBITS        00000000 001000 000694 00  AX  0   0  4
  [ 2] .sdata            PROGBITS        70000000 002000 000028 00  WA  0   0  4
  [ 3] .bss              NOBITS          70000028 002028 000004 00  WA  0   0  4
  [ 4] .comment          PROGBITS        00000000 002028 000012 01  MS  0   0  1
  [ 5] .riscv.attributes RISCV_ATTRIBUTES 00000000 00203a 000025 00      0   0  1
  [ 6] .shstrtab         STRTAB          00000000 00205f 000038 00      0   0  1
```

`riscv-example/src/main.c`:
```
#include <stdint.h>

volatile int global = 42;
volatile uint32_t controller_status = 0;

uint32_t tmp_1 = 0x12345678;
uint32_t tmp_2 = 0x00001100;
uint32_t tmp_3 = 0x00220000;
uint32_t tmp_4 = 0x33003300;
uint32_t tmp_5 = 0x00004400;
uint32_t tmp_6 = 0x00005500;
uint32_t tmp_7 = 0x00006600;
uint32_t tmp_8 = 0x77777777;

char hw[] = {
	'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd', '!', ' ', ' ',
	' ', 224, 251, 255, ' ', 252, 255, 249, 218, ' ', ' ', ' ', 21, 177,
	140, 130, 144, ' ', 29, 13, 6
};

volatile uint8_t *video_text_data_get(int x, int y) {
	return (volatile uint8_t *)(0x500FE800 + x + y * 64);
}

void print_hex(int x, int y, uint8_t n) {
	// Print a 0-15 number as hex value, '?' when error
	if (n < 10)
		*video_text_data_get(x, y) = n + '0';
	else if (n < 16)
		*video_text_data_get(x, y) = n - 10 + 'a';
	else
		*video_text_data_get(x, y) = '?';
}

void print_uint32(int x, int y, uint32_t n) {
	// Print uint32_t as hex value
	// Right is lower memory address
	for (int i = 0; i < 8; i++)
		print_hex(x + i, y, (n >> (32 - 4 - i * 4)) & 0xf);
}

volatile char *VIDEO_MEMORY = (volatile char *)(0x50000000 + 0xFE800);
int main() {
    int a = 4;
    int b = 12;
    int last_global = 42;
    int x_pos = 12;

    VIDEO_MEMORY[0] = 'H';
    VIDEO_MEMORY[1] = 'e';
    VIDEO_MEMORY[2] = 'l';
    VIDEO_MEMORY[3] = 'l';
    VIDEO_MEMORY[4] = 'o';
    VIDEO_MEMORY[5] = ' ';
    VIDEO_MEMORY[6] = 'W';
    VIDEO_MEMORY[7] = 'o';
    VIDEO_MEMORY[8] = 'r';
    VIDEO_MEMORY[9] = 'l';
    VIDEO_MEMORY[10] = 'd';
    VIDEO_MEMORY[11] = '!';
    VIDEO_MEMORY[12] = 'X';

	print_uint32(30, 30, (uint32_t) hw);	// line 63

	print_uint32(42, 17, (uint32_t) tmp_1);
	print_uint32(42, 18, (uint32_t) tmp_2);
	print_uint32(42, 19, (uint32_t) tmp_3);
	print_uint32(42, 20, (uint32_t) tmp_4);
	print_uint32(42, 21, (uint32_t) tmp_5);
	print_uint32(42, 22, (uint32_t) tmp_6);
	print_uint32(42, 23, (uint32_t) tmp_7);
	print_uint32(42, 24, (uint32_t) tmp_8);
	print_uint32(42, 25, (uint32_t) &tmp_1);

    while (1) {
        int c = a + b + global;
        if(global != last_global){
            if(controller_status){
                VIDEO_MEMORY[x_pos] = ' ';
                if(controller_status & 0x1){
                    if(x_pos & 0x3F){
                        x_pos--;
                    }
                }
                if(controller_status & 0x2){
                    if(x_pos >= 0x40){
                        x_pos -= 0x40;
                    }
                }
                if(controller_status & 0x4){
                    if(x_pos < 0x8C0){
                        x_pos += 0x40;
                    }
                }
                if(controller_status & 0x8){
                    if((x_pos & 0x3F) != 0x3F){
                        x_pos++;
                    }
                }
                VIDEO_MEMORY[x_pos] = 'X';
            }
            last_global = global;
        }
    }
    return 0;
}
```

**Comment by lxylxy123456** [comment]
There seems to be some alignment issue.
Simulator memory:
```
70000000: 48 65 6c 6c 6f 20 57 6f 72 6c 64 21 20 20 20 e0 Hello World!   .
70000010: fb ff 20 fc ff f9 da 20 20 20 15 b1 8c 82 90 20 .. ....   ..... 
70000020: 1d 0d 06 2a 1b 00 00 78 56 34 12 00 11 00 00 00 ...*...xV4......
70000030: 00 22 00 00 33 00 33 00 44 00 00 00 55 00 00 00 ."..3.3.D...U...
70000040: 66 00 00 77 77 77 77 00 e8 0f 50 00 00 00 00 00 f..wwww...P.....
70ffff90: 00 00 00 00 00 00 00 00 00 00 00 00 b4 ff ff 70 ...............p
70ffffa0: 00 00 00 00 00 00 00 00 8d 0a 00 00 00 00 00 00 ................
70ffffb0: 19 00 00 00 00 00 00 71 1a 00 00 78 10 00 00 00 .......q...x....
70ffffc0: 31 00 00 00 08 00 00 00 19 00 00 00 71 ee 0f 50 1...........q..P
70ffffd0: 00 00 00 00 00 00 00 00 00 00 00 00 74 03 00 00 ............t...
70ffffe0: 2b 00 00 78 0c 00 00 00 04 00 00 00 0c 00 00 00 +..x............
70fffff0: 1b 00 00 78 4c 00 00 70 00 00 00 71 24 00 00 00 ...xL..p...q$...
```
elf file (`hexdump -C`)
```
00002000  48 65 6c 6c 6f 20 57 6f  72 6c 64 21 20 20 20 e0  |Hello World!   .|
00002010  fb ff 20 fc ff f9 da 20  20 20 15 b1 8c 82 90 20  |.. ....   ..... |
00002020  1d 0d 06 00 2a 00 00 00  78 56 34 12 00 11 00 00  |....*...xV4.....|
00002030  00 00 22 00 00 33 00 33  00 44 00 00 00 55 00 00  |.."..3.3.D...U..|
00002040  00 66 00 00 77 77 77 77  00 e8 0f 50 47 43 43 3a  |.f..wwww...PGCC:|
00002050  20 28 47 4e 55 29 20 31  30 2e 32 2e 30 00 41 24  | (GNU) 10.2.0.A$|
00002060  00 00 00 72 69 73 63 76  00 01 1a 00 00 00 04 04  |...riscv........|
00002070  05 72 76 33 32 65 31 70  39 5f 6d 32 70 30 00 08  |.rv32e1p9_m2p0..|
00002080  01 0a 0b 00 2e 73 68 73  74 72 74 61 62 00 2e 74  |.....shstrtab..t|
00002090  65 78 74 00 2e 64 61 74  61 00 2e 73 64 61 74 61  |ext..data..sdata|
000020a0  00 2e 62 73 73 00 2e 63  6f 6d 6d 65 6e 74 00 2e  |..bss..comment..|
000020b0  72 69 73 63 76 2e 61 74  74 72 69 62 75 74 65 73  |riscv.attributes|
000020c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
```
Note this line:
```
00002020  1d 0d 06 00 2a 00 00 00  78 56 34 12 00 11 00 00  |....*...xV4.....|
70000020: 1d 0d 06 2a 1b 00 00 78 56 34 12 00 11 00 00 00 ...*...xV4......
                   ^^
```

**Comment by lxylxy123456** [comment]
Fixed because #8 is fixed indirectly
